interfaces/dotnet/Cantera/src/SpeciesCollection.cs

```diff
@@ -0,0 +1,172 @@
+using System.Buffers;
+using System.Collections;
+using Cantera.Interop;
+
+namespace Cantera;
+
+public class SpeciesCollection : IReadOnlyList<Species>
+{
+    readonly ThermoPhaseHandle _handle;
```

Comment:
So that’s a good question. The idea of the `SpeciesCollection` is two-fold:
1.  The `ThermoPhase` class is enormous; break it up a bit for better alignment with object-oriented best principles.
     - Any property belonging to the phase in bulk would be added to the `ThermoPhase` class.
     - And property like mass fractions which are set all-at-once for every item would be added to the `SpeciesCollection` class.
     - Any (read-only) property such as name which applies to the item would be `Species` class.
1.  It is very idiomatic in .NET world to have collections classes any time you have a bunch of items. These classes implement `IEnumerable` or a more specialized interface such as `IReadOnlyList`.

I realize that trying to make thermodynamics fit neatly into OO-world is futile, so my attempt here is to compromise between what Cantera‘s CLIB does and what a seasoned C# developer accustomed to business and web development might expect.